src/static/js/chapter.js

```diff
@@ -1,3 +1,61 @@
+//This function removes the lazy-loading attributes from all img and iframe tags
+//Useful for print view for example (https://bugs.chromium.org/p/chromium/issues/detail?id=875403)
+function removeLazyLoading() {
+
+  console.log("Removing lazy loading...");
+
+  var all_imgs = document.querySelectorAll('img');
+  for (index = 0; index < all_imgs.length; ++index) {
+    var img = all_imgs[index];
+    if (img.getAttribute('loading')) {
+      img.removeAttribute('loading');
+    }
+  }
+  
+  var all_iframes = document.querySelectorAll('iframe');
+  for (index = 0; index < all_iframes.length; ++index) {
+    var iframe = all_iframes[index];
+    if (iframe.getAttribute('loading')) {
+      iframe.removeAttribute('loading');
+    }
+  }
+}
+
+//Add an event handler to remove LazyLoading when entering print mode
+function removeLazyLoadingOnPrint() {
+  if ("onbeforeprint" in window) {
+    window.onbeforeprint = function() {
+      removeLazyLoading();
+    }
+  }
+
+}
+
+//Check if in print mode so we can remove lazy loading and block interactive visuals
+function printMode() {
+  var printMode = false;
+  var field = 'print';
+  var url = window.location.href;
+
+  if(url.indexOf('?' + field + '=') != -1) {

```

Comment:
I think this could be simplified using the URL API:

```js
var url = new URL(window.location);
var printMode = url.searchParams.has('print');
gtag('event', 'print-mode', { 'event_category': 'user', 'event_label': '' + printMode, 'value': +printMode })
return printMode;
```

Reply:
Again not IE11 compatible. IE11 isn’t a huge amount of usage but at same time don’t think it’s a massive improvement so would prefer to keep IE11 support for this is we can.

WDYT?

Reply:
```js
if (!('URL' in window)) {
  return false;
}
```

Gracefully degrade. IE11 users don't get print mode in that case. We should still be free to write clean, modern code whenever possible.

Reply:
Fair enough.

And they will get normal print mode when actually printing - just not URL param print mode - as they do support `onbeforeprint` and print stylesheets.